src/som/interpreter/Method.java

```diff
@@ -77,6 +80,20 @@ public boolean equals(final Object o) {
     return false;
   }
 
+  public void setNewCaller(final BackCacheCallNode caller) {
```

Comment:
You are changing a `@CompilationFinal` field here, do you need to invalidate any code that may have been compiled with this assumption backed in?

Reply:
Oh, I thought @CompilationFinal was taking care of that for me. Yes code needs to be invalidated.

Reply:
So I removed the @CompilationFinal of this variable. The problem is in BackCacheCallNode >> multipleCaller (). When this method is called, all the optimized code using the BackCacheCallNode has to be invalidated. I am not sure on how to do that.

Reply:
Not sure why I don't see these in the review, but we probably still want this to be `@CompilationFinal` for performance reasons.

Reply:
I tried to fix the concern with invalidation (the 4 comments) in my latest commit.

Since the field uniqueCaller is used in Method only during relinking, it is not compiled by Graal and does not have to be compilation final. The fields uniqueCaller in CachedDispatchNode and LexicallyBoundDispatchNode have however to be. I created an assumption at creation time which is invalidated when the method makeMultipleCaller (was: multipleCaller) is executed.

Looks like some of the test code I have still runs that way.